src/components/home.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import JSONTree from 'react-json-tree';
+
+import DivJSON from './div_json';
+
+import { createAuth, getTokens, renewIdToken, decodeIdToken, refreshSession, closeSession, signOut } from '../actions/index';
+
+class Home extends Component {
+  componentWillMount() {
+    //  create auth that goes in props
+    this.props.createAuth();
+  }
+
+  sessionClosed() {
+    return localStorage.getItem('session_closed');
+  }
+
+  getTokens() {
+    const { auth, login } = this.props;
+    this.props.getTokens(login, auth);
+  }
+
+  renewIdToken() {
+    const { auth } = this.props;
+    this.props.renewIdToken(auth);
+  }
+
+  decodeIdToken() {
+    const { auth, tokens } = this.props;
+    this.props.decodeIdToken(tokens.idToken, auth);
+  }
+
+  refreshSession() {
+    const { auth } = this.props;
+    this.props.refreshSession(auth);
+  }
+
+  closeSession() {
+    const { auth } = this.props;
+    localStorage.setItem('session_closed', true);
```

Comment:
yeah, rather than saving "session_closed" in localStorage, it probably makes more sense to save the positive case, i.e. idToken or accessTokens that we're using to communicate with Okta (depending on what you're showing and the expiration of the tokens). For example, for a simple app you probably don't want to have any state at all - when the user loads the page, you should almost always first ask Okta if a session exists and then get idTokens/etc off of the existing Okta session (if going through that flow). Otherwise, if you're storing the idToken, you should instead check its expiration to see if it's still valid or not.